### PR TITLE
Feature[C4R-713] Practice Program list and actions

### PR DESCRIPTION
**Pull request by birloiflorian**
- [C4R-713](https://wearetribus.atlassian.net/browse/C4R-713) Practice program list view with
- [C4R-764](https://wearetribus.atlassian.net/browse/C4R-764) Practice program actions (delete/edit/enable/disable)

### CODE REVIEW

File: frontend/src/pages/application/components/ApplicationDetails.tsx [hunk 1]
```diff
@@ -199,96 +206,73 @@ const ApplicationDetails = () => {
...
-              </div>
+              <ApplicationFeedbackCard
+                icon={<CheckCircleIcon className="text-green w-6" />}
+                title={t('details.active')}
+                description={t('details.define_active')}
+                actions={
+                  <>
+                    {!application?.pullingType && (
+                      <button
+                        className="save-button pl-8 pr-8 flex gap-4 sm:text-sm lg:text-base text-xs"
```
Comment (AndreeaMariaP) [inline frontend/src/pages/application/components/ApplicationDetails.tsx:217]: pl-8 pr-8 se pot scrie usor px-8, e mai simplu de modficat dupa daca e cazul (in loc sa modifici in 2 locuri, modifici in 1)

File: frontend/src/pages/pactice-program/components/PracticeProgramContent.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        </ShowMoreText>
+      </div>
+    </div>
+  );
+};
+
+const PracticeProgramContent = ({ program }: PracticeProgramContentProps) => {
+  const { t } = useTranslation('practice_program');
+
+  const calculatePerid = () => {
```
Comment (AndreeaMariaP) [inline frontend/src/pages/pactice-program/components/PracticeProgramContent.tsx:54]: typo calculatePerid

**Review by AndreeaMariaP — CHANGES_REQUESTED** [review]

File: backend/src/modules/application/entities/application.entity.ts [hunk 3]
```diff
@@ -24,6 +25,14 @@ export class Application extends BaseEntity {
   })
   status: ApplicationStatus;
 
+  @Column({
+    type: 'enum',
+    enum: ApplicationPullingType,
+    name: 'pulling_type',
+    nullable: true,
+  })
+  pullingType: ApplicationPullingType;
```
Comment (radulescuandrew) [inline backend/src/modules/application/entities/application.entity.ts:34]: managementModule, managementCeva, nu am deloc inspiratie 🗡️ 😄 

Practic asta e modul prin care vedem daca o aplicatie are o arie separata de management. 

`externalManagementType`, `externalConfigurationType`? nu am inspiratie deloc

File: frontend/src/pages/apps-store/components/AddApplicationConfig.ts [hunk 4]
```diff
@@ -54,6 +59,17 @@ const translations = {
   url: i18n.t('common:url'),
 };
 
+export const PullingTypeOptions = [
```
Comment (radulescuandrew) [inline frontend/src/pages/apps-store/components/AddApplicationConfig.ts:62]: ApplicationsExternalConfigurationZoneOptions? 😄 😆

File: frontend/src/services/application/Application.service.ts [hunk 5]
```diff
@@ -139,13 +139,18 @@ const generateApplicationFormDataPayload = (
...
+  const { steps, pullingType, ...data } = applicationDto;
   // create form data payload
   const payload = new FormData();
   for (const prop in cleanupPayload(data)) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     payload.append(prop, (data as any)[prop] as string);
   }
+
+  if (pullingType) {
+    payload.append('pullingType', pullingType.value);
```
Comment (radulescuandrew) [inline frontend/src/services/application/Application.service.ts:151]: de ce nu intra in `data` si asta?

**Review by radulescuandrew — CHANGES_REQUESTED** [review]

File: backend/src/modules/practice-program/constants/errors.constants.ts [hunk 6]
```diff
@@ -31,6 +31,10 @@ export const PRACTICE_PROGRAMS_ERRORS = {
     message: 'Error while searching practice programs',
     errorCode: 'PP_008',
   },
+  ENABLE_DISABLE: {
+    message: 'Error while updateing status',
```
[the remark this answers is not in the document]
Comment (birloiflorian) [reply backend/src/modules/practice-program/constants/errors.constants.ts]: Fixed.

File: frontend/src/pages/pactice-program/components/PracticeProgramContent.tsx (hunk 2 above)
Comment (birloiflorian) [reply frontend/src/pages/pactice-program/components/PracticeProgramContent.tsx]: Fixed

File: backend/src/modules/practice-program/services/practice-program.service.ts [hunk 7]
```diff
@@ -260,6 +260,48 @@ export class PracticeProgramService {
...
+    try {
+      const where = organizationId
+        ? {
+            id,
+            organizationId,
+          }
+        : { id };
+
+      const practiceProgram = await this.practiceProgramRepository.get({
+        where: where,
```
[the remark this answers is not in the document]
Comment (birloiflorian) [reply backend/src/modules/practice-program/services/practice-program.service.ts]: Fixed.

**Review by AndreeaMariaP — APPROVED** [review]

**Pull request merged**